include/deal.II/matrix_free/dof_info.templates.h

```diff
@@ -1664,188 +659,171 @@ namespace internal
             ++dat;
         }
       };
+
+      // We construct the connectivity graph in parallel. we use one lock for
+      // 256 degrees of freedom to keep the number of locks down to a
+      // reasonable level and reduce the cost of locking to some extent.
+      static constexpr unsigned int bucket_size_threading = 256;
```

Comment:
This block of changes (until line 759) parallelizes the construction of the connectivity graph (which is the most costly part in the construction of the thread parallelization scheme).